src/swap/components/SwapProvider.tsx

```diff
@@ -73,26 +94,10 @@ export function SwapProvider({
   useEffect(() => {
     // Error
     if (lifeCycleStatus.statusName === 'error') {
-      setLoading(false);
-      setPendingTransaction(false);
-      setError(lifeCycleStatus.statusData);
       onError?.(lifeCycleStatus.statusData);
     }
-    if (lifeCycleStatus.statusName === 'amountChange') {
-      setError(undefined);
-    }
-    if (lifeCycleStatus.statusName === 'transactionPending') {
-      setLoading(true);
-      setPendingTransaction(true);
-    }
-    if (lifeCycleStatus.statusName === 'transactionApproved') {
-      setPendingTransaction(false);
-    }
```

Comment:
no more reacting to state changes to set state

Reply:
so smoooooooth